### DISCUSSION TITLE
How to create multiple files concisely?

### DISCUSSION THREAD

**Discussion by ShaiAvr** [nushell/nushell#13608] [Q&A]
I am trying out NuShell, and I wanted to know how I can create multiple files in a concise way. In bash or zsh I can use brace expansion so that commands like `touch test{1..4}.txt` quickly creates `test1.txt`, `test2.txt`, `test3.txt` and `test4.txt`. I can also do `echo Hello > test{1..4}.txt` to create these files with the contents `"Hello"`. How can I do that in NuShell. Obviously, I can use a loop, but I wanted to know if there is a short and concise command like those I showed for bash/zsh.

**Comment by ghost** [comment]
nu does have `glob` (only existing files (bash `cat *.txt`)) and `str exapand` (bash `touch {1,2}.txt`), but its far from beeing compact.  
bash `touch {1,2}.txt` -> nu `touch ...("{1,2}.txt" | str expand)`  
bash `echo Hello > test{1..4}.txt` -> nu `"test{1..4}.txt" | str expand | each {|i| "Hello" | save $i}`

its a known weak point, but `>` for example will probably never be added to nu and auto-globbing is also controversal.  
statement from a dev on the matter and some workarounds: <https://github.com/nushell/nushell/discussions/9256>  
and in general more for the problem: <https://github.com/nushell/nushell/discussions?discussions_q=is%3Aopen+glob>
